build-scripts/package-tests.cake

```diff
@@ -0,0 +1,382 @@
+// Representation of a single test to be run against a pre-built package.
+// Each test has a Level, with the following values defined...
+//  0 Do not run - used for temporarily disabling a test
+//  1 Run for all CI tests - that is every time we test packages
+//  2 Run only on PRs, dev builds and when publishing
+//  3 Run only when publishing
```

Comment:
Is this functionality used? I think at this point it's probably overkill, right? Maybe we could simplify things for now?